spec/builtin-history.test.sh

```diff
@@ -195,4 +233,131 @@ status=1
 ## END
 
 
+#### HISTSIZE shrinks the history when changed
+
+cd $TMP
+printf "cmd %s\n" {1..10} > tmp
+HISTFILE=tmp
+history -c
+history -r
+history | wc -l
+HISTSIZE=5
+history | wc -l
+
+## STDOUT:
+10
+5
+## END
+
+
+#### HISTFILESIZE shrinks the history file when changed
+
+cd $TMP
+printf "cmd %s\n" {1..10} > tmp
+HISTFILE=tmp
+HISTFILESIZE=5
+cat tmp | wc -l
+
+## STDOUT:
+5
+## END
+
+
+#### set -o history / set +o history
+
+cd $TMP
+printf "echo %s\n" {1..3} > tmp
+HISTFILE=tmp $SH -i <<'EOF'
+set +o history
+echo "not recorded" >> /dev/null
+set -o history
+echo "recorded" >> /dev/null
+EOF
+
+case $SH in bash) echo '^D' ;; esac
+
+grep "not recorded" tmp >> /dev/null
+echo status=$?
+grep "recorded" tmp >> /dev/null
+echo status=$?
+
+## STDOUT:
+^D
+status=1
+status=0
+## END
+
+
+#### shopt histappend toggle check
+
+shopt -s histappend
+echo status=$?
+shopt -p histappend
+shopt -u histappend
+echo status=$?
+shopt -p histappend
+
+# match osh's behaviour of echoing ^D for EOF
+case $SH in bash) echo '^D' ;; esac
+
+## STDOUT:
+status=0
+shopt -s histappend
+status=0
+shopt -u histappend
+^D
+## END
+
+
+#### shopt histappend 
```

Comment:
I think this is the one where we might not want to match bash behavior, for simplicity

If so then let's make a note here

I think it makes sense to implement the better behavior ... It's possible people will notice the difference, but it's OK if they push back and we have to "concede"